configure.ac

```diff
@@ -262,6 +261,32 @@ LD="$saved_LD"
 # AUTOHARDEN END
 #####################################################
 
+#####################################################
+# OPTIMIZATION LEVEL START
+
+saved_CC="$CC"
+saved_LD="$LD"
+flag_wrap="$srcdir/wrap-compiler-for-flag-check"
+CC="$flag_wrap $CC"
+LD="$flag_wrap $LD"
+
+OPTIMIZATION_CFLAGS=""
+
+AS_IF([test x"$hardening" != x"yes"], [
+  check_cc_flag([-O3], [OPTIMIZATION_CFLAGS="$OPTIMIZATION_CFLAGS -O3"])
+  check_cc_flag([-g -ggdb3], [OPTIMIZATION_CFLAGS="$OPTIMIZATION_CFLAGS -g -ggdb3"])
```

Comment:
You generally want debug flags added only when building debug mode.

Reply:
Fixed